app/controllers/tickets_controller.rb

```diff
@@ -106,4 +106,189 @@ def edit_person_validators
+      }
+      return true
+    end
+
+    if user.nil? && person.nil?
+      render status: :unprocessable_entity, json: {
+        error: "User ID and WCA ID is not provided.",
+      }
+      return true
+    end
+    false
+  end
+
+  def details_before_anonymization
+    user, person = get_user_and_person
+    return if check_errors(user, person)
+
+    anonymization_checks = {}
+    message_args = {}
+    action_items = []
+    non_action_items = []
+
+    # 1. Check whether the user is currently banned or not.
+    anonymization_checks[:user_currently_banned] = user&.banned?
+
+    # 2. Check whether the user was banned in the past.
+    anonymization_checks[:user_banned_in_past] = user&.banned_in_past?
+
+    # 3. Check whether the person has held any records in the past.
+    records = person&.records
+    anonymization_checks[:person_has_records_in_past] = records.present? && records[:total] > 0
+    message_args[:records] = records
+
+    # 4. Check whether the person had podium on national championships.
+    championship_podiums = person&.championship_podiums
+    anonymization_checks[:person_held_championship_podiums] = championship_podiums&.values_at(:world, :continental, :national)&.any?(&:present?)
+    message_args[:championship_podiums] = championship_podiums
+
+    # 5. Check whether the person has competed in last 3 months.
+    recent_competitions_3_months = person&.competitions&.select { |c| c.start_date > (Date.today - 3.month) }
+    anonymization_checks[:person_competed_in_last_3_months] = recent_competitions_3_months&.any?
+    message_args[:recent_competitions_3_months] = recent_competitions_3_months
+
+    # 6. Check whether the user has account in WCA Forum.
+    anonymization_checks[:user_may_have_forum_account] = user.present?
+
+    # 7. Check whether the user has any active OAuth access grants.
+    access_grants = user&.oauth_access_grants&.select { |access_grant| !access_grant.revoked_at.nil? }
+    anonymization_checks[:user_has_active_oauth_access_grants] = access_grants&.any?
+    message_args[:access_grants] = access_grants
+
+    # 8. Check whether there are any competitions with external websites.
+    competitions_with_external_website = person&.competitions&.select { |c| c.external_website.present? }
+    anonymization_checks[:competitions_with_external_website] = competitions_with_external_website&.any?
+    message_args[:competitions_with_external_website] = competitions_with_external_website
+
+    # 9. Check whether there are any recent competitions data to be removed from WCA Live.
+    anonymization_checks[:recent_competitions_data_to_be_removed_wca_live] = recent_competitions_3_months&.any?
+
+    anonymization_checks.each { |key, value| (value ? action_items : non_action_items) << key }
+
+    render json: {
+      user: user,
+      person: person,
+      action_items: action_items,
+      non_action_items: non_action_items,
+      message_args: message_args,
+    }
+  end
+
+  def anonymize
```

Comment:
Again, definitely too much business logic happening inside the controller. Please move this into either a module in `lib` or into methods in `person`/`user`. Whatever seems appropriate to you

Reply:
As a rule of thumb, code along the lines of
```ruby
model_id = params.require(:id)
my_model = MyModel.find(model_id)

if some_condition
  my_model.update(foo: ..., bar: ...)
else
  my_model.delete
end

return status: :ok
```
is generally acceptable in a controller. Our codebase isn't the perfect rolemodel everywhere (especially keep your fingers from `competitions_controller` in that regard!) but your body here is definitely too much.